packages/mask/src/plugins/NextID/contexts/TipContext.tsx

```diff
@@ -0,0 +1,167 @@
+import { rightShift } from '@masknet/web3-shared-base'
+import {
+    ERC721ContractDetailed,
+    EthereumTokenType,
+    FungibleTokenDetailed,
+    isSameAddress,
+    TransactionState,
+    useNativeTokenDetailed,
+    useTokenConstants,
+    useTokenTransferCallback,
+} from '@masknet/web3-shared-evm'
+import { noop } from 'lodash-unified'
+import {
+    createContext,
+    Dispatch,
+    FC,
+    SetStateAction,
+    useCallback,
+    useContext,
+    useEffect,
+    useMemo,
+    useState,
+} from 'react'
+import { TipTask, TipType } from '../types'
+
+interface ContextOptions {
+    recipient: string
+    setRecipient: Dispatch<SetStateAction<string>>
+    tipType: TipType
+    setTipType: Dispatch<SetStateAction<TipType>>
+    recipients: string[]
+    setRecipients: Dispatch<SetStateAction<string[]>>
+    token: FungibleTokenDetailed | null
+    setToken: Dispatch<SetStateAction<FungibleTokenDetailed | null>>
+    amount: string
+    setAmount: Dispatch<SetStateAction<string>>
+    erc721TokenId: string | null
+    setErc721TokenId: Dispatch<SetStateAction<string | null>>
+    erc721Contract: ERC721ContractDetailed | null
+    setErc721Contract: Dispatch<SetStateAction<ERC721ContractDetailed | null>>
+}
+
+export const TipContext = createContext<ContextOptions>({
+    recipient: '',
+    setRecipient: noop,
+    tipType: TipType.NFT,
+    setTipType: noop,
+    recipients: [],
+    setRecipients: noop,
+    token: null,
+    setToken: noop,
+    amount: '',
+    setAmount: noop,
+    erc721TokenId: null,
```

Comment:
This plugin will support multi-chains. Better to avoid using chain-specific naming.